pkg/detector/preemption.go

```diff
@@ -0,0 +1,181 @@
+package detector
+
+import (
+	"context"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/klog/v2"
+
+	policyv1alpha1 "github.com/karmada-io/karmada/pkg/apis/policy/v1alpha1"
+	"github.com/karmada-io/karmada/pkg/features"
+	"github.com/karmada-io/karmada/pkg/util"
+	"github.com/karmada-io/karmada/pkg/util/helper"
+	"github.com/karmada-io/karmada/pkg/util/names"
+)
+
+// preemptionEnabled checks if preemption is enabled.
+func preemptionEnabled(preemption policyv1alpha1.PreemptionBehavior) bool {
+	if preemption != policyv1alpha1.PreemptAlways {
+		return false
+	}
+	if !features.FeatureGate.Enabled(features.PolicyPreemption) {
+		klog.Warningf("Cannot handle the preemption process because feature gate %q is not enabled.", features.PolicyPreemption)
+		return false
+	}
+	return true
+}
+
+// handleClusterPropagationPolicyPreemption handles the preemption process of PropagationPolicy.
+// If preemption is not enabled, the process will be aborted.
+// The preemption rule: high-priority PP > low-priority PP > CPP.
+func (d *ResourceDetector) handlePropagationPolicyPreemption(policy *policyv1alpha1.PropagationPolicy) error {
+	if preemptionEnabled(policy.Spec.Preemption) {
+		return nil
+	}
```

Comment:
This logic violates the rule `If preemption is not enabled, the process will be aborted.`.

Reply:
thanks for your find.